### PR TITLE
Fetch matches for a particular team only

### PR DESCRIPTION
**Pull request by SibiAkkash**
@Joe2k pushing the changes from #214 in separate PRs to make reviewing and rebasing easier

### CODE REVIEW

File: server/api.py [hunk 1]
```diff
@@ -1128,6 +1129,37 @@ def get_tournament_matches_by_slug(
...
+    except Team.DoesNotExist:
+        return 400, {"message": "Team does not exist"}
+
+    tournament_team_matches = Match.objects.filter(
+        Q(team_1=team) | Q(team_2=team), tournament=tournament
+    ).order_by("time")
+
+    end = time.time()
+
+    logging.debug(f"\nTime to get team matches: {(end - start)*1000}ms\n")
```
Comment (Joe2k) [inline server/api.py:1158]: can we remove this as its only to debug in local?

Comment (SibiAkkash) [reply server/api.py:1158]: sure, i wanted to check the time, we can take this after the tournaments with full data dump. Removing for now

Comment (SibiAkkash) [reply server/api.py]: Done

**Comment by SibiAkkash** [comment]
can you push this to main, so i'll rebase on this and push the skeletons PRs

**Comment by Joe2k** [comment]
yep one sec.. ill review fast and merge

**Comment by Joe2k** [comment]
@SibiAkkash merged

**Pull request merged**